src/modm/ui/color/concepts.hpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <bit>
+
+/**
+ * @brief		Concepts for colortypes and groups of colortypes.
+ * 				Key applications:
+ * 				- Color conversion constructors
+ * 				- Specialisation of buffer manipulation algorithms
+ *
+ * @author		Thomas Sommer
+ * @ingroup		modm_ui_color
+ */
+namespace modm::color {
+
+
+// Forward declarations
+template <int D>
+requires (D > 0)
+class GrayD;
```

Comment:
What does the `D` stand for in `GrayD`?

Reply:
RGBD: red, green, blue, depth (output of a depth camera like Realsense/...), so maybe GrayD is "color" of an b/w depth camera? 😉 😉 😜

(I know, D here probably is for _Number of Digits_.)
Why can't we just call it `Gray` instread of `GrayD` (same for `RgbD`/...)? Otherwise people using depth cameras and modm at the same time might get confused...

Reply:
... or W for Width: RgbW

Reply:
Okay... after more concideration, simple feels right -> Have removed the **D** ;)